### ISSUE TITLE
Dist size too large

### ISSUE THREAD

**Issue by Ricardo-Marques** [saasquatch/squatch-js#23]
We have build errors when a particular file in our dis is too large (>244kb, default webpack behavior)

When running webpack analyzer through our code to find out why it was failing, it seems squatch is the only offender.

![image](https://user-images.githubusercontent.com/9907599/94309316-7e94f280-ff3d-11ea-8bc0-b56c6a2b45c8.png)

It weighs in at 552.8KB.

I tried some basic tree shaking, by changing

```
import * as squatch from '@saasquatch/squatch-js'
```

to

```
import {init, widgets, api} from '@saasquatch/squatch-js'
```

But it seems like squatch is not tree shakable. [This is a good article on that subject.](https://medium.com/@craigmiller160/how-to-fully-optimize-webpack-4-tree-shaking-405e1c76038#:~:text=Just%20because%20Webpack%20can't,important%20impact%20on%20the%20application.&text=A%20file%20with%20side%20effects,the%20application%20as%20a%20whole.)

It would be nice to have a way of importing just the methods I need to save some precious KB.

**Comment by loganvolkers** [comment]
Hey @Ricardo-Marques thanks for the feedback.

I think the real problem here is that we only expose a UMD build, but not a CommonJS or ES Module build. You can visualize this via the webpack visualizer https://unpkg.com/@saasquatch/squatch-js@2.2.1/dist/stats.html to see that a lot of the size comes from superagent (our request library) and core-js. If we didn't need to support older browsers, we could cut both of those dependencies down.

We need to maintain a UMD build since that is one of the primary ways that customers use Squatch.js (that is directly in the browser instead of via a bundled build).

This is also visible if you look at the file sizes in `dist` -- there isn't a substantial size difference between the `squatch.WidgetApi.js` and `squatch.js`.

For you immediate problem switching to `squatch.min.js` might be the quick fix.

The longer fix would be exposing an ES Module build. Open to pull requests if you know how to get Webpack config right.

**Comment by Ricardo-Marques** [comment]
Makes sense. I'll give the min file a shot and will report back in the next couple business days. I may or may not look into the build config later.
